network/dell/dnos10_command.py

```diff
@@ -0,0 +1,208 @@
+        command. The task will wait for each condition to be true
+        before moving forward. If the conditional is not true
+        within the configured number of retries, the task fails.
+        See examples.
+    required: false
+    default: null
+    aliases: ['waitfor']
+  retries:
+    description:
+      - Specifies the number of retries a command should by tried
+        before it is considered failed. The command is run on the
+        target device every retry and evaluated against the
+        I(wait_for) conditions.
+    required: false
+    default: 10
+  interval:
+    description:
+      - Configures the interval in seconds to wait between retries
+        of the command. If the command does not pass the specified
+        conditions, the interval indicates how long to wait before
+        trying the command again.
+    required: false
+    default: 1
+"""
+
+EXAMPLES = """
+# Note: examples below use the following provider dict to handle
+#       transport and authentication to the node.
+vars:
+  cli:
+    host: "{{ inventory_hostname }}"
+    username: admin
+    password: admin
+    transport: cli
+
+tasks:
+  - name: run show version on remote devices
+    dnos10_command:
+      commands: show version
+      provider "{{ cli }}"
+
+  - name: run show version and check to see if output contains OS10
+    dnos10_command:
+      commands: show version
+      wait_for: result[0] contains OS10
+      provider "{{ cli }}"
+
+  - name: run multiple commands on remote nodes
+     dnos10_command:
+      commands:
+        - show version
+        - show interface
+      provider "{{ cli }}"
+
+  - name: run multiple commands and evaluate the output
+    dnos10_command:
+      commands:
+        - show version
+        - show interface
+      wait_for:
+        - result[0] contains OS10
+        - result[1] contains Ethernet
+      provider: "{{ cli }}"
+"""
+
+RETURN = """
+stdout:
+  description: The set of responses from the commands
+  returned: always
+  type: list
+  sample: ['...', '...']
+
+stdout_lines:
+  description: The value of stdout split into a list
+  returned: always
+  type: list
+  sample: [['...', '...'], ['...'], ['...']]
+
+failed_conditions:
+  description: The list of conditionals that have failed
+  returned: failed
+  type: list
+  sample: ['...', '...']
+
+warnings:
+  description: The list of warnings (if any) generated by module based on arguments
+  returned: always
+  type: list
+  sample: ['...', '...']
+"""
+from ansible.module_utils.basic import get_exception
+from ansible.module_utils.netcli import CommandRunner, FailedConditionsError
+from ansible.module_utils.dnos10 import NetworkModule, NetworkError
+
+def to_lines(stdout):
+    for item in stdout:
+        if isinstance(item, basestring):
+            item = str(item).split('\n')
+        yield item
+
+def main():
+    spec = dict(
+        commands=dict(type='list', required=True),
+        wait_for=dict(type='list', aliases=['waitfor']),
```

Comment:
@privateip Do you think we need to add the alias for `waitfor` on new modules?
I'm in two minds, 
On one hand it keeps it consistent across Networking Modules, on the other hand, this is a new module and therefore doesn't have any legacy issues